app/controllers/api/v2/template_combinations_controller.rb

```diff
@@ -1,25 +1,40 @@
 module Api
   module V2
     class TemplateCombinationsController < V2::BaseController
-      before_filter :find_resource, :only => [:show, :destroy]
+      before_filter :find_optional_nested_object
+      before_filter :find_resource, :only => [:show, :update, :destroy]
       before_filter :find_parent_config_template, :only => [:index, :create]
 
+      def_param_group :template_combination_identifiers do
+        param :config_template_id, String, :desc => N_("ID of config template")
+        param :hostgroup_id, String, :desc => N_("ID of host group")
+        param :environment_id, String, :desc => N_("ID of environment")
+      end
+
       api :GET, "/config_templates/:config_template_id/template_combinations", N_("List template combination")
-      param :config_template_id, :identifier, :required => true
+      api :GET, "/hostgroups/:hostgroup_id/template_combinations", N_("List template combination")
+      api :GET, "/environments/:environment_id/template_combinations", N_("List template combination")
+      param_group :template_combination_identifiers
       def index
-        @template_combinations = @config_template.template_combinations
+        @template_combinations = requested_resource.template_combinations
         @total = @template_combinations.count
       end
 
-      api :POST, "/config_templates/:config_template_id/template_combinations", N_("Add a template combination")
-      param :config_template_id, :identifier, :required => true
-      param :template_combination, Hash, :required => true do
-        param :environment_id, :number, :allow_nil => true, :desc => N_("environment id")
-        param :hostgroup_id, :number, :allow_nil => true, :desc => N_("host group id")
+      def_param_group :template_combination do
+        param :template_combination, Hash, :required => true, :action_aware => true do
+          param :environment_id, :number, :allow_nil => true, :desc => N_("environment id")
+          param :hostgroup_id, :number, :allow_nil => true, :desc => N_("host group id")
+        end
       end
 
+      api :POST, "/config_templates/:config_template_id/template_combinations", N_("Add a template combination")
+      api :POST, "/hostgroups/:hostgroup_id/template_combinations", N_("Add a template combination")
+      api :POST, "/environments/:environment_id/template_combinations", N_("Add a template combination")
```

Comment:
These routes don't seem to work, they're not listed in `rake routes` for me and I get: `ActionController::RoutingError (No route matches [POST] "/api/v2/environments/production/template_combinations")`